src/main/java/io/cdap/plugin/gcp/dataplex/common/config/DataplexBaseConfig.java

```diff
@@ -0,0 +1,242 @@
+import io.cdap.plugin.gcp.common.GCPConnectorConfig;
+import io.cdap.plugin.gcp.common.GCPUtils;
+import io.cdap.plugin.gcp.dataplex.common.util.DataplexConstants;
+
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import java.io.IOException;
+
+import javax.annotation.Nullable;
+
+/**
+ * Contains common Dataplex config properties.
+ */
+public class DataplexBaseConfig extends PluginConfig {
+  protected static final String REFERENCE_NAME = "referenceName";
+  protected static final String NAME_LAKE = "lake";
+  protected static final String NAME_ZONE = "zone";
+  protected static final String NAME_LOCATION = "location";
+  protected static final String NAME_CONNECTION = "connection";
+  private static final Logger LOG = LoggerFactory.getLogger(DataplexBaseConfig.class);
+
+  @Name(REFERENCE_NAME)
+  @Description("Name used to uniquely identify this plugin for lineage, annotating metadata, etc.")
+  protected String referenceName;
+
+  @Name(NAME_LOCATION)
+  @Macro
+  @Description("Resource id for the Dataplex location. On the lake page, click the name of the lake. " +
+    "The lake page has the location ID.")
+  protected String location;
+
+  @Name(NAME_LAKE)
+  @Macro
+  @Description("Resource id for the Dataplex lake. You can find it on the lake detail page in Dataplex.")
+  protected String lake;
+
+  @Name(NAME_ZONE)
+  @Macro
+  @Description("Resource id for the Dataplex zone. You can find it on the zone detail page in Dataplex.")
+  protected String zone;
+
+  @Name(NAME_CONNECTION)
+  @Nullable
+  @Macro
+  @Description("The existing connection to use.")
+  protected GCPConnectorConfig connection;
+
+  public String getReferenceName() {
+    return referenceName;
+  }
+
+  public String getLake() {
+    return lake;
+  }
+
+  public void setLake(String lake) {
+    this.lake = lake;
+  }
+
+  public String getZone() {
+    return zone;
+  }
+
+  public void setZone(String zone) {
+    this.zone = zone;
+  }
+
+  public String getLocation() {
+    return location;
+  }
+
+  /**
+   * Return true if the service account is set to auto-detect but it can't be fetched from the environment.
+   * This shouldn't result in a deployment failure, as the credential could be detected at runtime if the pipeline
+   * runs on dataproc. This should primarily be used to check whether certain validation logic should be skipped.
+   *
+   * @return true if the service account is set to auto-detect but it can't be fetched from the environment.
+   */
+  public boolean autoServiceAccountUnavailable() {
+    if (connection == null || connection.getServiceAccountFilePath() == null &&
+      connection.isServiceAccountFilePath()) {
+      try {
+        ServiceAccountCredentials.getApplicationDefault();
+      } catch (IOException e) {
+        return true;
+      }
+    }
+    return false;
+  }
+
+  /**
+   * @return GoogleCredentials based on service account
+   */
+  public GoogleCredentials getCredentials() {
```

Comment:
Can we reuse `connection.getCredentials()`? This will reduce code duplication.

Reply:
No, it doesn't handle the auto-detect scenerio while getting credentials

Reply:
So for `auto-detect` we can just add an additional check if this method returned `null` and `service account file path` was set to detect instead of duplicating the method. Does that makes sense?